### DISCUSSION TITLE
Keypress stuff

### DISCUSSION THREAD

**Discussion by onetribegamingcodes** [CommandPost/CommandPost#3081] [Technical Support] [answered]
Just a random question i had, is there a way that you can hold any keypress when the midi note is on and have it release when its off??

**Comment by latenitefilms** [comment, marked as answer]
We have "press and hold" and "release" actions already for the modifier keys. If you want non-modifier keys though, you'd need to make your own "press" and "release" Lua Snippets. For the "a" key for example:

**Press:**
```lua
hs.eventtap.event.newKeyEvent("a", true):post()
```

**Release:**
```lua
hs.eventtap.event.newKeyEvent("a", false):post()
```

See: http://www.hammerspoon.org/docs/hs.eventtap.event.html#newKeyEvent

**Reply by onetribegamingcodes** [reply]
ah arlight, thank you for the help

**Reply by onetribegamingcodes** [reply]
ive managed to get it to hold, but how can i get it to release the key when i release an midi note?

**Reply by onetribegamingcodes** [reply]
arlight nevermind i got it to work
